test/include/materials/LinearNonLinearIterationInterfaceMaterial.h

```diff
@@ -0,0 +1,36 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#pragma once
+
+#include "InterfaceMaterial.h"
+
+// Forward Declarations
+class LinearNonLinearIterationInterfaceMaterial;
```

Comment:
If we can remove the legacy validParams, then we can also remove this forward declaration